components/brave_rewards/browser/rewards_notification_service_observer_extensions_impl.cc

```diff
@@ -0,0 +1,131 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/browser/rewards_notification_service_observer_extensions_impl.h"
+
+#include "brave/common/extensions/api/rewards_notifications.h"
+#include "chrome/browser/profiles/profile.h"
+#include "extensions/browser/event_router.h"
+
+namespace brave_rewards {
+
+RewardsNotificationServiceObserverExtensionsImpl::
```

Comment:
so typically `Impl` is used when we have a single implementation of an interface, like RewardsService/RewardsServiceImpl. For an observer the pattern would be something like `ExtensionRewardsNotificationServiceObserver`